cmd/kubeops/internal/handler/dashboardcompat.go

```diff
@@ -0,0 +1,100 @@
+// The Dashboard doesn't understand the Helm 3 release format.
+// This file is a compatibility layer that translates Helm 3 releases to a Helm 2-similar format suitable for the Dashboard.
+
+package handler
+
+import (
+	"strings"
+
+	"github.com/golang/protobuf/ptypes/any"
+	h3chart "helm.sh/helm/v3/pkg/chart"
+	h3 "helm.sh/helm/v3/pkg/release"
+	h2chart "k8s.io/helm/pkg/proto/hapi/chart"
+	h2 "k8s.io/helm/pkg/proto/hapi/release"
+)
+
+type dashboardCompatibleRelease struct {
+	Name      string                    `json:"name,omitempty"`
+	Info      h2.Info                   `json:"info,omitempty"`
+	Chart     dashboardCompatibleChart  `json:"chart,omitempty"`
+	Config    dashboardCompatibleConfig `json:"config,omitempty"`
+	Manifest  string                    `json:"manifest,omitempty"`
+	Version   int                       `json:"version,omitempty"`
+	Namespace string                    `json:"namespace,omitempty"`
+}
+
+type dashboardCompatibleChart struct {
+	Files     []*any.Any                `json:"files,omitempty"`
+	Metadata  h3chart.Metadata          `json:"metadata,omitempty"`
+	Templates []*h2chart.Template       `json:"templates,omitempty"`
+	Values    dashboardCompatibleConfig `json:"values,omitempty"`
+}
+
+type dashboardCompatibleConfig struct {
+	Raw string `json:"raw,omitempty"`
+}
+
+func newDashboardCompatibleRelease(h3r h3.Release) dashboardCompatibleRelease {
+	return dashboardCompatibleRelease{
+		Name:      h3r.Name,
+		Info:      h2.Info{Status: compatibleStatus(*h3r.Info)},
+		Chart:     compatibleChart(*h3r.Chart),
+		Config:    compatibleConfig(*h3r.Chart),
+		Manifest:  h3r.Manifest,
+		Version:   h3r.Version,
+		Namespace: h3r.Namespace,
+	}
+}
+
+func compatibleChart(h3c h3chart.Chart) dashboardCompatibleChart {
+	return dashboardCompatibleChart{
+		Files:     compatibleFiles(h3c.Files),
+		Metadata:  *h3c.Metadata,
+		Templates: compatibleTemplates(h3c.Templates),
+		Values:    compatibleConfig(h3c),
```

Comment:
According to comments in the Helm code, "Config is the set of extra Values added to the chart. These values override the default values inside of the chart.". I guess that means that we should not use `compatibleConfig` to set the values. `Config` be the subset of Values that we saw somewhere (you know that yaml that was stripped of comments...), but we should also have a `compatibleValues()` that returns the _actual_ full Values contents.

https://github.com/helm/helm/blob/master/pkg/release/release.go#L29-L31

Reply:
You're probably right.

Reply:
It seems like `config` is the final, modified `values`. I deployed `bitnami/apache` with the `replicaCount` and `metric.resources.limits` properties manually modified in the GUI, and then compared `values.raw` with `config.raw` (both from the JSON response):

```diff
$ diff json-values.yaml json-config.yaml 
27d26
< 
51d49
< 
55d52
< 
58c55
< replicaCount: 1
---
> replicaCount: 5
86d82
< 
90d85
< 
94d88
< 
98d91
< 
164d156
< 
170d161
< 
179c170
<   secrets:
---
>   secrets: 
225,227c216,218
<     limits: {}
<     #   cpu: 100m
<     #   memory: 128Mi
---
>     limits:
>         cpu: 100m
>         memory: 128Mi
```

Note that both `values.raw` and `config.raw` preserved all YAML comments.